examples/hello.cpp

```diff
@@ -0,0 +1,111 @@
+#include <cstdio>
+
+#include <js/Initialization.h>
+#include <jsapi.h>
+
+/* This example illustrates the bare minimum you need to do to execute a
+ * JavaScript program using embedded SpiderMonkey. It does no error handling and
+ * simply exits if something goes wrong.
+ *
+ * To use the interpreter you need to create a context and a global object, and
+ * do some setup on both of these. You also need to enter a "request" (lock on
+ * the interpreter) and a "compartment" (environment within one global object)
+ * before you can execute code. */
+
+// clang-format off
+static JSClassOps globalOps = {
+  nullptr, // addProperty
+  nullptr, // deleteProperty
+  nullptr, // enumerate
+  nullptr, // newEnumerate
+  nullptr, // resolve
+  nullptr, // mayResolve
+  nullptr, // finalize
+  nullptr, // call
+  nullptr, // hasInstance
+  nullptr, // construct
+  JS_GlobalObjectTraceHook
+};
+
+// The class of the global object.
+static JSClass globalClass = {
+  "HelloWorldGlobal",
+  JSCLASS_GLOBAL_FLAGS,
+  &globalOps
+};
+// clang-format on
+
+static JSContext*
+CreateContext(void)
+{
+  JSContext* cx = JS_NewContext(8L * 1024 * 1024);
+  if (!cx)
+    return nullptr;
+  if (!JS::InitSelfHostedCode(cx))
+    return nullptr;
+  return cx;
+}
+
+static JSObject*
+CreateGlobal(JSContext* cx)
+{
+  JS::CompartmentOptions options;
+  JS::RootedObject global(
+    cx,
+    JS_NewGlobalObject(
+      cx, &globalClass, nullptr, JS::FireOnNewGlobalHook, options));
+
+  // Add standard JavaScript classes to the global so we have a useful
+  // environment.
+  JSAutoCompartment ac(cx, global);
+  if (!JS_InitStandardClasses(cx, global))
+    return nullptr;
+
+  return global;
+}
+
+static bool
+ExecuteCodePrintResult(JSContext* cx, const char* code)
+{
+  JS::CompileOptions options(cx);
+  options.setFileAndLine("noname", 1);
+  JS::RootedValue rval(cx);
+  if (!JS::Evaluate(cx, options, code, strlen(code), &rval))
+    return false;
+
+  printf("%s\n", JS_EncodeString(cx, rval.toString()));
```

Comment:
Not quite accurate, I think; that would be `JS::ToString()` if I understand correctly. I'll put something like "In this case, we know that the value is a string because of the expression that we executed, so we can just get the string directly."